donut/modules/groups/helpers.py

```diff
@@ -82,3 +82,45 @@ def get_group_data(group_id, fields=None):
     # Return the row in the form of a dict
     result = {f: t for f, t in zip(fields, result)}
     return result
+
+
+
+def get_position_data(fields=None):
+    all_returnable_fields = ["group_id", "pos_id", "pos_name"]
+    default_fields = ["pos_name", "pos_id"]
+
+    if fields is None:
+        fields = default_fields
+    else:
+        if any(f not in all_returnable_fields for f in fields):
+            return "Invalid field"
+
+    s = sqlalchemy.sql.select(fields).select_from(sqlalchemy.text("positions"))
+    
+    result = flask.g.db.execute(s)
+
+    if result is None:
+        return {}
+    
+    res = {}
+    for row in result:
+        res[row["pos_id"]] = row["pos_name"]
+    
+
+    fields = ["user_id", "pos_id", "group_id"]
+    s = sqlalchemy.sql.select(fields).select_from(sqlalchemy.text("position_holders"))
```

Comment:
Rather than run 2 queries, you could just join the two tables `WHERE positions.pos_id = position_holders.pos_id`